### DISCUSSION TITLE
Compare 'exact match' Economy models with randomly sampled models

### DISCUSSION THREAD

**Discussion by sbenthall** [econ-ark/HARK#1668] [General]
It looks like some of the Economy models use an 'exact match' technique when forward-simulating some random shocks, such as employment shocks in KrusellSmith. The number of employed people is kept constant based on the economy state and permuted, rather than sampled based on an individualized Markov process.

As we have discussed many times, in most cases this technique maintains the expected values of the simulation, approximating more closely what we would get from an "infinite" version of the model, while reducing the variance.

Forward simulation of agents is not used in the solution of _micro_ problems in HARK. But for _macro_ problems that use the Market class, forward simulation code is used as part of the solution method. The model is forward simulated a number of times in order to get equilibrium expectations.

Because the agents are generally risk averse, I suspect that the exact match sampling in the forward simulation of these models may be throwing off the results. Wider variance in outcomes (based on wider variance in, for example, employment) should effect individual consumption choices.

I could be wrong. But it seems peculiar that the correctness of this technique is assumed in some models and not tested explicit. This would be easy enough to do: provide both options and run simulation tests to compare them explicitly.

**Comment by llorracc** [comment]
Agents _expectations_ are unbiased, so the simulations do not violate their expectations to a first order.  The problems with them are really pretty minor.  I'd guess that there's a literature about this somewhere.  As long as the fiction is maintained that we are really trying to simulate the population of an economy with arbitrarily many consumers, it's hard to see how it could go badly wrong -- thought it would be nice to have some citations.

**Comment by sbenthall** [comment]
This seems like an unanswered research question.

**Comment by sbenthall** [comment]
@nicksawhney maybe a notebook to explore this?
